### ISSUE TITLE
Quantum View v2 Returning 503 Service Unavailable

### ISSUE THREAD

**Issue by rbur0425** [UPS-API/api-documentation#121]
First, I get the access token - 

```
public function createAccessToken()
    {
        $curl = curl_init();

        $payload = "grant_type=client_credentials";

        curl_setopt_array($curl, [
            CURLOPT_HTTPHEADER => [
                "Content-Type: application/x-www-form-urlencoded",
                "x-merchant-id: " . $this->merchant_id,
                "Authorization: Basic " . base64_encode($this->client_id . ":" . $this->client_secret)
            ],
            CURLOPT_POSTFIELDS => $payload,
            CURLOPT_URL => "https://onlinetools.ups.com/security/v1/oauth/token",
            CURLOPT_RETURNTRANSFER => true,
            CURLOPT_CUSTOMREQUEST => "POST",
            CURLOPT_SSL_VERIFYPEER => true,
            CURLOPT_SSL_VERIFYHOST => 2,
            CURLOPT_TIMEOUT => 120,
        ]);

        // Execute the request and handle the response
        $response = curl_exec($curl);

        if (curl_errno($curl)) {
            Log::error('Curl error: ' . curl_error($curl));
            throw new Exception("Error getting UPS token");
        } else {
            $httpcode = curl_getinfo($curl, CURLINFO_HTTP_CODE);
            Log::info('HTTP Status Code: ' . $httpcode);
            Log::info('Response: ' . $response);
        }

        curl_close($curl);

        // Decode the response
        $responseDecoded = json_decode($response);

        return $responseDecoded->access_token;
    }
```

Then I send the QVEvents request with the token I just got  - 

```
public function BuildRequest($beginDateTime, $endDateTime, $bookmark = null, $accessToken)
    {
        $curl = curl_init();

        $payload = array(
            "QuantumViewRequest" => array(
                "Request" => array(
                    "RequestAction" => "QVEvents"
                ),
                "SubscriptionRequest" => array(
                    "Name" => "Third Party Shipments",
                    "DateTimeRange" => array(
                        "BeginDateTime" => $beginDateTime,
                        "EndDateTime" => $endDateTime
                    )
                )
            )
        );

        if ($bookmark !== null) {
            $payload["QuantumViewRequest"]["Bookmark"] = $bookmark;
        }

        curl_setopt_array($curl, [
            CURLOPT_HTTPHEADER => [
                "Authorization: Bearer " . $accessToken,
                "Content-Type: application/json"
            ],
            CURLOPT_POSTFIELDS => json_encode($payload),
            CURLOPT_URL => "https://onlinetools.ups.com/api/quantumview/v2/events",
            CURLOPT_RETURNTRANSFER => true,
            CURLOPT_CUSTOMREQUEST => "POST",
            CURLOPT_SSL_VERIFYPEER => true,
            CURLOPT_SSL_VERIFYHOST => 2,
            CURLOPT_TIMEOUT => 120,
            CURLOPT_VERBOSE => true,
            CURLOPT_ENCODING => '',
        ]);

        $maxRetries = 3;
        $retryDelay = 1;

        for ($retry = 0; $retry < $maxRetries; $retry++) {
            $response = curl_exec($curl);
            $error = curl_error($curl);
            $httpCode = curl_getinfo($curl, CURLINFO_HTTP_CODE);

            if ($httpCode === 503) {
                if ($retry < $maxRetries - 1) {
                    dump("Retrying in {$retryDelay} seconds.");
                    sleep($retryDelay);
                    $retryDelay *= 3; // Exponential backoff
                }
            } else {
                break;
            }
        }



        curl_close($curl);

        if ($error) {
            Log::error('Curl error: ' . $error);
            throw new Exception("Error getting Quantum View Events");
        }


        return json_decode($response);
    }
```
I get the JSON Response

```
{
    "response": {
        "errors": [
            {
                "code": "10004",
                "message": "The service is temporarily unavailable"
            }
        ]
    }
}
```

I receive the below verbose curl output (token removed for obvious reasons) 

* Found bundle for host: 0x60000267ce10 [can multiplex]
* Re-using existing connection with host onlinetools.ups.com
* [HTTP/2] [5] OPENED stream for https://onlinetools.ups.com/api/quantumview/v2/events
* [HTTP/2] [5] [:method: POST]
* [HTTP/2] [5] [:scheme: https]
* [HTTP/2] [5] [:authority: onlinetools.ups.com]
* [HTTP/2] [5] [:path: /api/quantumview/v2/events]
* [HTTP/2] [5] [accept: */*]
* [HTTP/2] [5] [accept-encoding: deflate, gzip, br, zstd]
* [HTTP/2] [5] [authorization: Bearer <TOKEN>]
* [HTTP/2] [5] [content-type: application/json]
* [HTTP/2] [5] [content-length: 200]
> POST /api/quantumview/v2/events HTTP/2
Host: onlinetools.ups.com
Accept: */*
Accept-Encoding: deflate, gzip, br, zstd
Authorization: Bearer <TOKEN>
Content-Type: application/json
Content-Length: 200

* upload completely sent off: 200 bytes
< HTTP/2 503
< content-type: application/json
< authtransid:
< errorcode: 500
< errordescription: GatewayTimeout
< x-frame-options: DENY
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< access-control-max-age: 600
< x-request-id: 98d26145-88c3-4690-8c4d-09de3a8ee0a3
< content-length: 114
< expires: Thu, 06 Jun 2024 02:22:22 GMT
< cache-control: max-age=0, no-cache, no-store
< pragma: no-cache
< date: Thu, 06 Jun 2024 02:22:22 GMT
< ak-grn-1: 0.6924c317.1717640522.30258429
< strict-transport-security: max-age=31536000 ; includeSubDomains
<
* Connection #0 to host onlinetools.ups.com left intact

**Comment by s6alboer** [comment]
Same issue.

I use QV and tracking for multiple customers. Before june this was just a slight hickup. The behavior got serious in the week of the anounced deadline. Now it is virtually impossible to do a successful QV request for some customers. 

I tried to narrow the date/time range, which helped, but didn't solve the problem.

I also noticed long response times for tracking (multiple seconds) and especially QV (up to 70seconds). It appears, that the service is lacking some compute power or IO.
